src/extensions/default/InlineBoxModelEditor/BoxModelEditor.js

```diff
@@ -66,17 +66,37 @@ define(function(require, exports, module) {
         });
     }
 
-    PaddingValue.prototype.toString = function() {
+    BoxModelValue.prototype.toString = function() {
         return this.value + (this.value === 0 ? "" : this.unit);
     };
 
-    function PaddingEditor($parent, valueString, paddingChangeHandler) {
+    function BoxModelEditor($parent, valueString, BoxModelChangeHandler, type) {
         var self = this;
 
+        if(type === "margin"){
+          BoxModelTemplate = BoxModelTemplate.replace("{{SET_PADDING_FOR}}", "{{SET_MARGIN_FOR}}");
```

Comment:
don't use string replace on mustache templates as it can be quite error-prone.

Instead, you just need to modify the strings passed into the render function like so:
```js
var headerKey = "";

switch (type) {
    case BoxModelUtils.MARGIN:
        headerKey = "SET_PADDING_FOR";
        break;
    case BoxModelUtils.PADDING:
        headerKey = "SET_MARGIN_FOR";
        break;
}

Strings["BOX_MODEL_HEADER"] = Strings[headerKey];

self.$element = $(Mustache.render(BoxModelTemplate, Strings));
```

Notice how I use `BoxModelUtils.MARGIN` and the corresponding one for padding as well? Make that change too (move those type constants to BoxModelUtils) so that we don't run into typo errors in constants.